swervelib/ctre/CanCoderFactoryBuilder.java

```diff
@@ -1,70 +1,91 @@
-import com.ctre.phoenix.sensors.WPI_CANCoder;
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.MagnetSensorConfigs;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.signals.AbsoluteSensorRangeValue;
+import com.ctre.phoenix6.signals.SensorDirectionValue;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import frc.team670.mustanglib.swervelib.AbsoluteEncoder;
 import frc.team670.mustanglib.swervelib.AbsoluteEncoderFactory;
 
 public class CanCoderFactoryBuilder {
 
-    private Direction direction = Direction.COUNTER_CLOCKWISE;
     private int periodMilliseconds = 10;
 
     public CanCoderFactoryBuilder withReadingUpdatePeriod(int periodMilliseconds) {
         this.periodMilliseconds = periodMilliseconds;
         return this;
     }
 
-    public CanCoderFactoryBuilder withDirection(Direction direction) {
-        this.direction = direction;
-        return this;
-    }
-
     public AbsoluteEncoderFactory<CanCoderAbsoluteConfiguration> build() {
         return configuration -> {
-            CANCoderConfiguration config = new CANCoderConfiguration();
-            config.absoluteSensorRange = AbsoluteSensorRange.Unsigned_0_to_360;
-            config.magnetOffsetDegrees = Math.toDegrees(configuration.getOffset());
-            config.sensorDirection = direction == Direction.CLOCKWISE;
-            config.initializationStrategy = configuration.getInitStrategy();
+            CANcoderConfiguration config = new CANcoderConfiguration();
+            CANcoder encoder = new CANcoder(configuration.getId(), configuration.getCanbus());
+            
+            
+            StatusCode val = encoder.getConfigurator().refresh(config);
+
+            if(val.isOK()){
+                  val = encoder.getConfigurator().refresh(config);
+            }
+
+            
+            config.MagnetSensor.AbsoluteSensorRange = AbsoluteSensorRangeValue.Unsigned_0To1;
+            // config.MagnetSensor.MagnetOffset = configuration.getOffset();// Should be roations
+            config.MagnetSensor.SensorDirection = SensorDirectionValue.CounterClockwise_Positive;
 
-            WPI_CANCoder encoder = new WPI_CANCoder(configuration.getId(), configuration.getCanbus());
-            CtreUtils.checkCtreError(encoder.configAllSettings(config, 250), "Failed to configure CANCoder");
+            encoder.getConfigurator().apply(config);
+            CtreUtils.checkCtreError(encoder.getConfigurator().apply(config), "Failed to configure CANCoder");
 
-            CtreUtils.checkCtreError(encoder.setStatusFramePeriod(CANCoderStatusFrame.SensorData, periodMilliseconds, 250), "Failed to configure CANCoder update rate");
+            // CtreUtils.checkCtreError(encoder.optimizeBusUtilization(periodMilliseconds), "Failed to configure CANCoder update rate");
```

Comment:
Please see my other comment on this PR for pointers to the functions in the new APIs that allow doing what the previous line 37 was doing